VirtoCommerce.PricingModule.Data/Handlers/DeletePricesProductChangedEvent.cs

```diff
@@ -0,0 +1,48 @@
+using System.Linq;
+using System.Threading.Tasks;
+using VirtoCommerce.Domain.Catalog.Events;
+using VirtoCommerce.Domain.Catalog.Model;
+using VirtoCommerce.Domain.Common.Events;
+using VirtoCommerce.Domain.Pricing.Model.Search;
+using VirtoCommerce.Domain.Pricing.Services;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Events;
+
+namespace VirtoCommerce.PricingModule.Data.Handlers
+{
+    public class DeletePricesProductChangedEvent : IEventHandler<ProductChangedEvent>
+    {
+
+        private readonly IPricingService _pricingService;
+        private readonly IPricingSearchService _pricingSearchService;
+
+        public DeletePricesProductChangedEvent(IPricingService pricingService, IPricingSearchService pricingSearchService)
+        {
+            _pricingService = pricingService;
+            _pricingSearchService = pricingSearchService;
+        }
+
+        public virtual Task Handle(ProductChangedEvent message)
+        {
+
+            foreach (var changedEntry in message.ChangedEntries.ToList())
+            {
+                DeletePrices(changedEntry);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private void DeletePrices(GenericChangedEntry<CatalogProduct> changedEntry)
```

Comment:
Please make this method as `protected` and `virtual` to allow support of future extensions